### PR TITLE
Misc. trivial typo fixes

### PR DESCRIPTION
**Pull request by luzpaz**
Typos found via `codespell`. Typos discovered are mainly trivial source comment typos. There is an occasional important typo fix.

### CODE REVIEW

File: sympy/combinatorics/perm_groups.py [hunk 1]
```diff
@@ -1958,7 +1958,7 @@ def _number_blocks(blocks):
                 to_remove = []
                 for i, r in enumerate(rep_blocks):
                     if len(r) > len(rep) and rep.issubset(r):
-                        # i-th block sytem is not minimal
+                        # i-th block system is not minimal
```
Comment (luzpaz) [inline sympy/combinatorics/perm_groups.py]: I'm not certain this fix is correct. Please review

File: sympy/combinatorics/partitions.py [hunk 2]
```diff
@@ -318,7 +318,7 @@ def __new__(cls, partition, integer=None):
         >>> IntegerPartition({1:3, 2:1})
         IntegerPartition(5, (2, 1, 1, 1))
 
-        If the value that the partion should sum to is given first, a check
+        If the value that the portion should sum to is given first, a check
```
Comment (luzpaz) [inline sympy/combinatorics/partitions.py]: I'm not sure this fix is correct, please review.

File: sympy/utilities/iterables.py [hunk 3]
```diff
@@ -1068,7 +1068,7 @@ def multiset_permutations(m, size=None, g=None):
 
 def _partition(seq, vector, m=None):
     """
-    Return the partion of seq as specified by the partition vector.
+    Return the portion of seq as specified by the partition vector.
```
Comment (luzpaz) [inline sympy/utilities/iterables.py]: I'm not sure this fix is correction either, please review

File: sympy/utilities/enumerative.py [hunk 4]
```diff
@@ -36,7 +36,7 @@
 part is greater than B's leftmost part.  If the leftmost parts are
 equal, compare the second parts, and so on.
 
-In this ordering, the greatest partion of a given multiset has only
+In this ordering, the greatest portion of a given multiset has only
```
Comment (luzpaz) [inline sympy/utilities/enumerative.py]: I'm not sure this fix is correct either, please review

File: sympy/combinatorics/partitions.py (hunk 2 above)
Comment (asmeurer) [reply sympy/combinatorics/partitions.py]: I believe these should be "partition".

File: sympy/core/multidimensional.py [hunk 5]
```diff
@@ -23,7 +23,7 @@ def apply_on_element(f, args, kwargs, n):
         structure = kwargs[n]
         is_arg = False
 
-    # Define reduced function that is only dependend of the specified argument.
+    # Define reduced function that is only dependent of the specified argument.
```
Comment (asmeurer) [inline sympy/core/multidimensional.py]: Should be "dependent on".

File: sympy/printing/fcode.py [hunk 6]
```diff
@@ -409,7 +409,7 @@ def split_pos_code(line, endpos):
                 if pos == 0:
                     return endpos
             return pos
-        # split line by line and add the splitted lines to result
+        # split line by line and add the split lines to result
```
Comment (asmeurer) [inline sympy/printing/fcode.py]: Is this change correct?

**Comment by asmeurer** [comment]
I went through all the changes and they all look good except for those comments.

File: sympy/printing/fcode.py (hunk 6 above)
Comment (luzpaz) [reply sympy/printing/fcode.py]: @asmeurer https://www.quora.com/Why-is-the-word-splitted-not-grammatically-correct

**Comment by luzpaz** [comment]
@asmeurer made the appropriate changes per your review. Repushed PR. Thanks you.

**Review by gxyd — APPROVED** [review]
I am generally not inclined towards PR's like these, which don't focus on changes to one section of code. But as Aaron has given +1. I'm merging these.

Also we tend to be a little negative towards force push.

Thanks for your contribution.

**Comment by asmeurer** [comment]
Spelling fixes are fine. Unlike code style fixes, spelling is well agreed upon, and a spelling change is unlikely to have unintended consequences.

**Pull request merged**